### PR TITLE
gh/commits

### PR DESCRIPTION
**Pull request by geneh**
- Add new request types for pull request commit & pull request commit comments.
- Process synchronize PullRequestEvent not present in the timeline which caused missing PR commits.

### CODE REVIEW

File: lib/visitorMap.js [hunk 1]
```diff
@@ -151,6 +151,20 @@ const commit = {
...
+const pull_request_commit_comment = {
+  _type: 'pull_request_commit_comment',
+  user: self
+};
+
+
+const pull_request_commit = {
+  _type: 'pull_request_commit',
+  pull_request_commit_comments: collection(pull_request_commit_comment),
+  repo: self,
```
Comment (willbar) [inline lib/visitorMap.js]: We might have discussed this but is it possible to link to the ```pull_request```? We typically only link one level up the hierarchy, so we would link to ```pull_request``` not ```repo```.

File: providers/fetcher/githubProcessor.js [hunk 2]
```diff
@@ -248,14 +254,23 @@ class GitHubProcessor {
...
+
+  commit_comment(request, isPullRequest = false) {
     // TODO links to consider
     // * reactions -- get this by using the following Accept header: application/vnd.github.squirrel-girl-preview
     const document = request.document;
     const context = request.context;
     request.addSelfLink();
-    request.linkResource('commit', context.qualifier);
-    request.linkSiblings(`${context.qualifier}:commit_comments`);
+    if (isPullRequest) {
```
Comment (willbar) [inline providers/fetcher/githubProcessor.js]: I prefer the pattern you did above where you used a variable containing the prefix and just conditioned setting that variable.

File: lib/visitorMap.js [hunk 3]
```diff
@@ -151,6 +151,19 @@ const commit = {
...
+const pull_request_commit_comment = {
+  _type: 'pull_request_commit_comment',
+  user: self
+};
+
+
+const pull_request_commit = {
+  _type: 'pull_request_commit',
+  pull_request_commit_comments: collection(pull_request_commit_comment),
+  author: self,
```
Comment (jeffmcaffer) [inline lib/visitorMap.js]: can/should we have a `repo` edge here?  like there is in the normal `commit`

File: providers/fetcher/githubProcessor.js [hunk 4]
```diff
@@ -981,8 +1000,8 @@ class GitHubProcessor {
     request.linkResource(relation.origin, `${qualifier}`);
     request.linkSiblings(`${relation.qualifier}:pages`);
     request.linkCollection('unique', `${relation.qualifier}:pages:${relation.guid}`);
-    const urnPrefix = relation.type === 'commit' && qualifier.includes('pull_request') ? `repo:${qualifier.split(':')[2]}:pull_request_commit` : relation.type;
-    const id = relation.type === 'commit' ? 'sha' : 'id';
+    const urnPrefix = relation.type === 'pull_request_commit' ? `repo:${qualifier.split(':')[2]}:pull_request_commit` : relation.type;
```
Comment (jeffmcaffer) [inline providers/fetcher/githubProcessor.js]: Interesting, pullreqeust commits are no longer listed in a relation (see change on line 300 (in new revision).  So do we even need this test any more?

**Pull request merged**